src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -310,7 +310,7 @@ abs_request_creator::make_list_blobs_request(
     }
 
     if (max_keys) {
-        target += fmt::format("&max_results={}", max_keys.value());
+        target += fmt::format("&maxresults={}", max_keys.value());
```

Comment:
I find it strange that we don't use continuation token in here

> If the request doesn't specify maxresults, or specifies a value greater than 5,000, the server will return up to 5,000 items. 
> —https://learn.microsoft.com/en-us/rest/api/storageservices/list-blobs?tabs=microsoft-entra-id

This can be truncated too.

Reply:
A bomb waiting to explode. Let's throw an exception if we get back a continuation token from so that truncated results are not dropped silently.

Reply:
Sorry if I'm not following, but wouldn't adding the [continuation token (called `marker` in the ABS API)](https://learn.microsoft.com/en-us/rest/api/storageservices/list-blobs?tabs=microsoft-entra-id#uri-parameters) back to `abs_client::do_list_objects()` and therefore handling truncation in the same manner as the `s3_client` be the ideal solution instead of throwing an exception?

We already handling the response parsing correctly in `abs_parse_impl::handle_start/end_element()`, so the `continuation_token` is set correctly, we just aren't passing it back in `abs_client::do_list_objects()` currently.

Reply:
That would be ideal of course. 🙂‍↕️

I was trying to suggest way out from implementing that in this PR by at least eliminating the footgun in the interim.

Reply:
It's not a big lift, will fix this issue in this PR.